pycaret/containers/models/time_series/time_series_clustering.py

```diff
@@ -0,0 +1,672 @@
+required parameters in the `__init__` and then call `super().__init__` to complete
+the process. Refer to the existing classes for examples.
+"""
+
+import logging
+import random
+import warnings
+from abc import abstractmethod
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import numpy as np  # type: ignore
+import pandas as pd
+from packaging import version
+from sktime.forecasting.base import BaseForecaster  # type: ignore
+from sktime.forecasting.compose import (  # type: ignore
+    TransformedTargetForecaster,
+    make_reduction,
+)
+from sktime.forecasting.trend import PolynomialTrendForecaster  # type: ignore
+from sktime.transformations.series.detrend import (  # type: ignore
+    ConditionalDeseasonalizer,
+    Detrender,
+)
+from sktime.transformations.series.summarize import WindowSummarizer
+
+import pycaret.containers.base_container
+from pycaret.containers.models.base_model import (
+    ModelContainer,
+    leftover_parameters_to_categorical_distributions,
+)
+from pycaret.internal.distributions import (
+    CategoricalDistribution,
+    Distribution,
+    IntUniformDistribution,
+    UniformDistribution,
+)
+from pycaret.utils._dependencies import _check_soft_dependencies
+from pycaret.utils.datetime import (
+    coerce_datetime_to_period_index,
+    coerce_period_to_datetime_index,
+)
+from pycaret.utils.generic import get_logger, np_list_arange, param_grid_to_lists
+from pycaret.utils.time_series import TSModelTypes
+from pycaret.utils.time_series.forecasting.models import _check_enforcements
+
+# First one in the list is the default ----
+ALL_ALLOWED_ENGINES: Dict[str, List[str]] = {
+    "auto_arima": ["pmdarima", "statsforecast"],
+    "lr_cds_dt": ["sklearn", "sklearnex"],
+    "en_cds_dt": ["sklearn", "sklearnex"],
+    "ridge_cds_dt": ["sklearn", "sklearnex"],
+    "lasso_cds_dt": ["sklearn", "sklearnex"],
+    "lar_cds_dt": ["sklearn"],
+    "llar_cds_dt": ["sklearn"],
+    "br_cds_dt": ["sklearn"],
+    "huber_cds_dt": ["sklearn"],
+    "par_cds_dt": ["sklearn"],
+    "omp_cds_dt": ["sklearn"],
+    "knn_cds_dt": ["sklearn", "sklearnex"],
+    "dt_cds_dt": ["sklearn"],
+    "rf_cds_dt": ["sklearn"],
+    "et_cds_dt": ["sklearn"],
+    "gbr_cds_dt": ["sklearn"],
+    "ada_cds_dt": ["sklearn"],
+    "xgboost_cds_dt": ["sklearn"],
+    "lightgbm_cds_dt": ["sklearn"],
+    "catboost_cds_dt": ["sklearn"],
+    # "svm_cds_dt": ["sklearn", "sklearnex"],
+}
+
+
+def get_container_default_engines() -> Dict[str, str]:
+    """Get the default engines from all models
+
+    Returns
+    -------
+    Dict[str, str]
+        Default engines for all containers. If unspecified, it is not included
+        in the return dictionary.
+    """
+    default_engines = {}
+    for id, all_engines in ALL_ALLOWED_ENGINES.items():
+        default_engines[id] = all_engines[0]
+    return default_engines
+
+
+class TimeSeriesKMeansContainer(ModelContainer):
```

Comment:
- You would need `TimeSeriesClusteringContainer` (which inherits from `ModelContainer`).
- Then all clustering model containers would inherit from this. e.g. class `TimeSeriesKMeansContainer(TimeSeriesClusteringContainer)`
- You can refer to the non-time series clustering examples for reference here if needed

https://github.com/pycaret/pycaret/blob/2cee9735d5dec12cb66d9baca27d7e82ccaf13c8/pycaret/containers/models/clustering.py#L46

https://github.com/pycaret/pycaret/blob/2cee9735d5dec12cb66d9baca27d7e82ccaf13c8/pycaret/containers/models/clustering.py#L181